python/ray/train/v2/api/data_parallel_trainer.py

```diff
@@ -31,24 +30,33 @@
 from ray.train.v2._internal.util import construct_train_func
 from ray.train.v2.api.config import RunConfig, ScalingConfig
 from ray.train.v2.api.result import Result
+from ray.util.annotations import Deprecated, DeveloperAPI
 from ray.util.scheduling_strategies import NodeAffinitySchedulingStrategy
 
 logger = logging.getLogger(__name__)
 
 
-TRAINER_RESTORE_DEPRECATION_WARNING = """
-The `Trainer.restore` API is deprecated and will be removed in a future release.
+_TRAINER_RESTORE_DEPRECATION_WARNING = """
+The `restore` and `can_restore` APIs are deprecated and
+will be removed in a future release.
 
-This API previously accepted a Train run directory path and loaded state such as the
-training code and configurations from a pkl file, which was hard to use.
-Now, Ray Train attempts to load the snapshot of reported checkpoints if it exists
-in the run directory, which makes `ray.train.get_checkpoint` available as long as
-you're pointing to the same run directory (i.e. the same `storage_path` and `name`).
+This API previously accepted a Train run directory path and
+loaded state (including user training code) partially from .pkl file and
+partially from new arguments passed to the constructor.
+This was confusing and error-prone.
 
-If you want to start a new training run without any prior checkpoint history, please
-specify a new, unique `RunConfig(name)`.
+Now, trainers only have a single constructor codepath, which takes in
```

Comment:
Should I have these long messages in the error itself? Or should I just have a short message + link to the migration guide with more details?

Reply:
Would probably recommend linking to a guide, but we can do this for now and update to point to the migration guide once we have it.